tests/telemetry/test_writer.py

```diff
@@ -45,37 +44,28 @@ def test_add_event(telemetry_writer, test_agent_session, mock_time):
 
 def test_add_event_disabled_writer(telemetry_writer, test_agent_session):
     """asserts that add_event() does not create a telemetry request when telemetry writer is disabled"""
-    initial_event_count = len(test_agent_session.get_requests())
-    telemetry_writer.disable()
-
     payload = {"test": "123"}
     payload_type = "test-event"
     # ensure events are not queued when telemetry is disabled
     telemetry_writer.add_event(payload, payload_type)
 
     # ensure no request were sent
-    telemetry_writer.periodic()
-    assert len(test_agent_session.get_requests()) == initial_event_count
+    telemetry_writer.periodic(force_flush=True)
+    assert len(test_agent_session.get_requests(payload_type)) == 1
 
 
 def test_app_started_event(telemetry_writer, test_agent_session, mock_time):
     """asserts that _app_started_event() queues a valid telemetry request which is then sent by periodic()"""
     with override_global_config(dict(_telemetry_dependency_collection=False)):
-        initial_event_count = len(test_agent_session.get_events())
         # queue an app started event
         telemetry_writer._app_started_event()
         # force a flush
-        telemetry_writer.periodic()
+        telemetry_writer.periodic(force_flush=True)
```

Comment:
Is adding `force_flush=True` meant to be the refresh state between tests?

Reply:
We use telemetry_writer.reset_queues() to reset the state between tests. This is done in the telemetryagentsession fixture. 


By default the periodic writer thread is configured to run every 10 seconds. On every call periodic aggregates metrics and logs and on the six invocation (on 1 minute intervals) it submits telemetry events to the agent (these intervals can be overridden via envars).

Here we call periodic with forced_flushed=True to ensure periodic always sends all queued telemetry events.